app/controllers/user_projects_controller.rb

```diff
@@ -0,0 +1,27 @@
+class UserProjectsController < ApplicationController
+  before_action :authenticate_user!
+  before_action :set_project
+
+  def assign
+    begin
+      @user_project = UserProject.create!(user: current_user, project: @project)
+      redirect_to @project, notice: "Project successfully assigned."
+    rescue
+      redirect_to projects_url, notice: "Assignment failed - unable to save record in database."
+    end
+  end
+
+  def leave
+    if @user_project = UserProject.find_by(user: current_user, project: @project)
```

Comment:
I'm not sure if this variable will be needed for views :smiley: however I can change it ;>